terraform-google-{{cookiecutter.module_name}}/test/make.sh

```diff
@@ -72,18 +72,13 @@ function docker() {
 
 # This function runs 'terraform validate' and 'terraform fmt'
 # against all directory paths which contain *.tf files.
-function check_terraform() {
-  set -e
-  echo "Running terraform validate"
+function check_terraform() {set -e
+  echo "Running terraform validate and terraform fmt"
   find_files . -name "*.tf" -print0 \
     | compat_xargs -0 -n1 dirname \
     | sort -u \
-    | compat_xargs -t -n1 terraform validate --check-variables=false
-  echo "Running terraform fmt"
-  find_files . -name "*.tf" -print0 \
-    | compat_xargs -0 -n1 dirname \
-    | sort -u \
-    | compat_xargs -t -n1 terraform fmt -check=true -write=false
+    | compat_xargs -t -n1 -i{} bash -c \
+    'terraform init "{}" > /dev/null && terraform validate "{}" && terraform fmt -check=true -write=false "{}"'
```

Comment:
This results in the behavior of re-initializing .terraform in the root of the module repository.  This happens over and over again affecting any initialization the user has done. 

Was this intended as a performance optimization to avoid initializing .terraform in each of the root module directories found?  If so, re-initializing /.terraform takes ~36 seconds total whereas changing the working directory to each root module takes ~39 seconds total, so not much of a difference.  Suggest changing this to use a `tests/terraform_validate` script similar to [helpers/terraform_validate](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/6a98ed5318dc1f13df007a167489b3eb7c4b2ef2/helpers/terraform_validate) because [the docs](https://www.terraform.io/docs/commands/init.html#usage) mention, "It is recommended to run Terraform with the current working directory set to the root directory of the configuration, and omit the DIR argument."

In addition, `terraform validate --help` advises including `-backend=false` to avoid re-initializing the backend.

```
  Validation requires an initialized working directory with any referenced
  plugins and modules installed. To initialize a working directory for
  validation without accessing any configured remote backend, use:
      terraform init -backend=false
```

Finally, terraform fmt should be separated out to make it clear what needs to be fixed.  `-diff` should be added to make clear what differences need to be corrected, otherwise it fails silently.

Reply:
I thought it would be useful to perform validation and formatting checks on each module in turn, rather than looping through each module twice to perform each action separately, but I will defer to the suggested implementation as it is better thought out.